modules/gib.js

```diff
@@ -0,0 +1,217 @@
+
+    if (handicap >= 8) {
+        points.add([mid, 1 + d])
+        points.add([mid, boardsize - d])
+    }
+
+    return points
+}
+
+exports.parse = function (input) {
+
+    // Try UTF-8 encoding... in any case it's better than binary.
+
+    input = iconv.decode(Buffer.from(input, 'binary'), 'utf8')
+
+    let lines = input.split('\n')
+
+    let tree = gametree.new()
+    let root = {}
+    tree.nodes.push(root)
+
+    root.FF = ['4']
+    root.GM = ['1']
+    root.SZ = ['19']
+
+    let node = root
+
+    for (let n = 0; n < lines.length; n++) {
+
+        let line = lines[n].trim()
+
+        if (line.startsWith('\\[GAMEBLACKNAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PB = [s]
+
+        } else if (line.startsWith('\\[GAMEWHITENAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PW = [s]
+
+        } else if (line.startsWith('\\[GAMECONDITION=')) {
+
+            // Hard-coded the common komi cases.
+            // For better results, we could do a regex here instead.
+
+            if (line.toLowerCase().includes('black 6.5 dum')) {
+                root.KM = ['6.5']
+            } else if (line.toLowerCase().includes('black 7.5 dum')) {
+                root.KM = ['7.5']
+            } else if (line.toLowerCase().includes('black 0.5 dum')) {
+                root.KM = ['0.5']
+            }
+
+        } else if (line.startsWith('\\[GAMERESULT=')) {
+
+            let score = null
+            let strings = line.split(' ')
+
+            // Try to find score by assuming any float found is the score.
+
+            for (let i in strings) {
+                try {
+                    let p = parseFloat(strings[i])  // Does this ever throw?
+                    if (isNaN(p) === false) {
+                        score = p
+                    }
+                }
+                catch (e) {}
+            }
+
+            if (line.toLowerCase().includes('white') &&
+                line.toLowerCase().includes('black') === false) {
+
+                if (line.toLowerCase().includes('resignation')) {
+                    root.RE = ['W+R']
+                } else if (score !== null) {
+                    root.RE = ['W+' + score.toString()]
+                } else {
+                    root.RE = ['W+']
+                }
+            }
+
+            if (line.toLowerCase().includes('black') &&
+                line.toLowerCase().includes('white') === false) {
+
+                if (line.toLowerCase().includes('resignation')) {
+                    root.RE = ['B+R']
+                } else if (score !== null) {
+                    root.RE = ['B+' + score.toString()]
+                } else {
+                    root.RE = ['B+']
+                }
+            }
+
+        } else if (line.slice(0, 3) === 'INI') {
+
+            let setup = line.split(' ')
+            let handicap = 0
+            try {
+                let p = parseInt(setup[3])
```

Comment:
Avoid using `parseInt()` without radix; it can lead to unexpected results such as `parseInt('080') == 0` in older browsers. [More information here.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) Just use `parseFloat()` instead (and `Math.round()` if necessary).